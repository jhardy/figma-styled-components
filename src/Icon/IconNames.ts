export type IconNames = 'AlignBottom' | 'AlignTop' | 'AlignVerticalCenter' | 'ArrowDown' | 'ArrowLeft' | 'ArrowRight' | 'ArrowUp' | 'CaretDown' | 'CaretRight' | 'CaretUp' | 'MenuCheckmarkBreadcrumb' | 'MenuCheckmarkMixed' | 'MenuCheckmarkOn' | 'MenuChevron' | 'Noodle' | 'OrientationLandscape' | 'OrientationPortrait' | 'Overlay' | 'Swap' | 'TextAlignAdjusted' | 'TextAlignCenter' | 'TextAlignLeft' | 'TextAlignRight' | 'TextDecorationStrikethrough' | 'TextDecorationUnderline' | 'TextResizeFixed' | 'TextResizeHeight' | 'TextResizeWidth' | 'TextTransformCapitalize' | 'TextTransformLower' | 'TextTransformUpper' | 'Adjust' | 'Angle' | 'ArrowLeftRight' | 'ArrowUpDown' | 'Blend' | 'BlendEmpty' | 'Break' | 'Component' | 'CornerRadius' | 'Corners' | 'DistributeHorizontalSpacing' | 'DistributeVerticalSpacing' | 'Draft' | 'Effects' | 'Eyedropper' | 'Frame' | 'Hidden' | 'Hyperlink' | 'Kebab' | 'LayoutAlignBottom' | 'LayoutAlignHorizontalCenters' | 'LayoutAlignLeft' | 'LayoutAlignRight' | 'LayoutAlignTop' | 'LayoutAlignVerticalCenters' | 'LayoutGridColumns' | 'LayoutGridRows' | 'LayoutGridUniform' | 'LetterSpacing' | 'Library' | 'LineHeight' | 'LinkBroke' | 'LinkConnected' | 'List' | 'ListDetailed' | 'Lock' | 'LockUnlocked' | 'Minus' | 'NodeDelete' | 'ParagraphIndent' | 'ParagraphSpacing' | 'Play' | 'Plus' | 'Recent' | 'ResetInstance' | 'ResizeToFit' | 'Resolve' | 'ResolveFilled' | 'Return' | 'Search' | 'SearchLarge' | 'Share' | 'Smiley' | 'StrokeWeight' | 'Styles' | 'TidyUpGrid' | 'TidyUpListHorizontal' | 'TidyUpListVertical' | 'Timer' | 'Trash' | 'VectorHandles' | 'Visible' | 'Warning' | 'X' | 'AnimateFill' | 'Group' | 'ImageFill' | 'Instance' | 'LayerComponent' | 'LayerFrame' | 'Mask' | 'ScrollingFrameAll' | 'ScrollingFrameHorizontal' | 'ScrollingFrameVertical' | 'Slice' | 'Type' | 'Vector'