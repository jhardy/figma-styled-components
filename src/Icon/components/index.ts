export { default as AlignBottom } from './16/AlignBottom'
export { default as AlignTop } from './16/AlignTop'
export { default as AlignVerticalCenter } from './16/AlignVerticalCenter'
export { default as ArrowDown } from './16/ArrowDown'
export { default as ArrowLeft } from './16/ArrowLeft'
export { default as ArrowRight } from './16/ArrowRight'
export { default as ArrowUp } from './16/ArrowUp'
export { default as CaretDown } from './16/CaretDown'
export { default as CaretRight } from './16/CaretRight'
export { default as CaretUp } from './16/CaretUp'
export { default as MenuCheckmarkBreadcrumb } from './16/MenuCheckmarkBreadcrumb'
export { default as MenuCheckmarkMixed } from './16/MenuCheckmarkMixed'
export { default as MenuCheckmarkOn } from './16/MenuCheckmarkOn'
export { default as MenuChevron } from './16/MenuChevron'
export { default as Noodle } from './16/Noodle'
export { default as OrientationLandscape } from './16/OrientationLandscape'
export { default as OrientationPortrait } from './16/OrientationPortrait'
export { default as Overlay } from './16/Overlay'
export { default as Swap } from './16/Swap'
export { default as TextAlignAdjusted } from './16/TextAlignAdjusted'
export { default as TextAlignCenter } from './16/TextAlignCenter'
export { default as TextAlignLeft } from './16/TextAlignLeft'
export { default as TextAlignRight } from './16/TextAlignRight'
export { default as TextDecorationStrikethrough } from './16/TextDecorationStrikethrough'
export { default as TextDecorationUnderline } from './16/TextDecorationUnderline'
export { default as TextResizeFixed } from './16/TextResizeFixed'
export { default as TextResizeHeight } from './16/TextResizeHeight'
export { default as TextResizeWidth } from './16/TextResizeWidth'
export { default as TextTransformCapitalize } from './16/TextTransformCapitalize'
export { default as TextTransformLower } from './16/TextTransformLower'
export { default as TextTransformUpper } from './16/TextTransformUpper'
export { default as Adjust } from './32/Adjust'
export { default as Angle } from './32/Angle'
export { default as ArrowLeftRight } from './32/ArrowLeftRight'
export { default as ArrowUpDown } from './32/ArrowUpDown'
export { default as Blend } from './32/Blend'
export { default as BlendEmpty } from './32/BlendEmpty'
export { default as Break } from './32/Break'
export { default as Component } from './32/Component'
export { default as CornerRadius } from './32/CornerRadius'
export { default as Corners } from './32/Corners'
export { default as DistributeHorizontalSpacing } from './32/DistributeHorizontalSpacing'
export { default as DistributeVerticalSpacing } from './32/DistributeVerticalSpacing'
export { default as Draft } from './32/Draft'
export { default as Effects } from './32/Effects'
export { default as Eyedropper } from './32/Eyedropper'
export { default as Frame } from './32/Frame'
export { default as Hidden } from './32/Hidden'
export { default as Hyperlink } from './32/Hyperlink'
export { default as Kebab } from './32/Kebab'
export { default as LayoutAlignBottom } from './32/LayoutAlignBottom'
export { default as LayoutAlignHorizontalCenters } from './32/LayoutAlignHorizontalCenters'
export { default as LayoutAlignLeft } from './32/LayoutAlignLeft'
export { default as LayoutAlignRight } from './32/LayoutAlignRight'
export { default as LayoutAlignTop } from './32/LayoutAlignTop'
export { default as LayoutAlignVerticalCenters } from './32/LayoutAlignVerticalCenters'
export { default as LayoutGridColumns } from './32/LayoutGridColumns'
export { default as LayoutGridRows } from './32/LayoutGridRows'
export { default as LayoutGridUniform } from './32/LayoutGridUniform'
export { default as LetterSpacing } from './32/LetterSpacing'
export { default as Library } from './32/Library'
export { default as LineHeight } from './32/LineHeight'
export { default as LinkBroke } from './32/LinkBroke'
export { default as LinkConnected } from './32/LinkConnected'
export { default as List } from './32/List'
export { default as ListDetailed } from './32/ListDetailed'
export { default as Lock } from './32/Lock'
export { default as LockUnlocked } from './32/LockUnlocked'
export { default as Minus } from './32/Minus'
export { default as NodeDelete } from './32/NodeDelete'
export { default as ParagraphIndent } from './32/ParagraphIndent'
export { default as ParagraphSpacing } from './32/ParagraphSpacing'
export { default as Play } from './32/Play'
export { default as Plus } from './32/Plus'
export { default as Recent } from './32/Recent'
export { default as ResetInstance } from './32/ResetInstance'
export { default as ResizeToFit } from './32/ResizeToFit'
export { default as Resolve } from './32/Resolve'
export { default as ResolveFilled } from './32/ResolveFilled'
export { default as Return } from './32/Return'
export { default as Search } from './32/Search'
export { default as SearchLarge } from './32/SearchLarge'
export { default as Share } from './32/Share'
export { default as Smiley } from './32/Smiley'
export { default as StrokeWeight } from './32/StrokeWeight'
export { default as Styles } from './32/Styles'
export { default as TidyUpGrid } from './32/TidyUpGrid'
export { default as TidyUpListHorizontal } from './32/TidyUpListHorizontal'
export { default as TidyUpListVertical } from './32/TidyUpListVertical'
export { default as Timer } from './32/Timer'
export { default as Trash } from './32/Trash'
export { default as VectorHandles } from './32/VectorHandles'
export { default as Visible } from './32/Visible'
export { default as Warning } from './32/Warning'
export { default as X } from './32/X'
export { default as AnimateFill } from './Layer/AnimateFill'
export { default as Group } from './Layer/Group'
export { default as ImageFill } from './Layer/ImageFill'
export { default as Instance } from './Layer/Instance'
export { default as LayerComponent } from './Layer/LayerComponent'
export { default as LayerFrame } from './Layer/LayerFrame'
export { default as Mask } from './Layer/Mask'
export { default as ScrollingFrameAll } from './Layer/ScrollingFrameAll'
export { default as ScrollingFrameHorizontal } from './Layer/ScrollingFrameHorizontal'
export { default as ScrollingFrameVertical } from './Layer/ScrollingFrameVertical'
export { default as Slice } from './Layer/Slice'
export { default as Type } from './Layer/Type'
export { default as Vector } from './Layer/Vector'