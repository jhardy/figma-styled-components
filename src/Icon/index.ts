export * from './Icon'
export * from './IconNames'
