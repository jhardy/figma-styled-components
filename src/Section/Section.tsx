import * as React from 'react'
import styled from 'styled-components'
import { SectionTitle } from '../SectionTitle'

export interface SectionProps {
  title: string
  actions?: JSX.Element
  collapsed?: boolean
}



class SectionFactory extends React.Component<SectionProps,{ collapsed: boolean}> {
  constructor (props: SectionProps) {
    super(props)
    this.state = {
      collapsed: this.props.collapsed || false
    }
  }

  public render () {
    return(
      <SectionWrapper collapsed={this.state.collapsed} {...this.props}>
          <SectionHeader>
            <SectionTitle>{this.props.title}</SectionTitle>
            {this.props.actions &&
              <SectionActions>{this.props.actions}</SectionActions>
            }
          </SectionHeader>
          <SectionContent>
            {this.props.children}
          </SectionContent>
      </SectionWrapper>
    )
  }
}

const SectionContent = styled.div``

const SectionActions = styled.div``

const SectionHeader = styled.div`
  display: flex;
  justify-content: space-between;
`

const SectionWrapper = styled.section<SectionProps>`
  padding: 8px;

  ${SectionHeader} {
    opacity: ${(props) => props.collapsed ? 0.5 : 1 };
  }

  ${SectionContent} {
    display: ${(props) => props.collapsed ? 'none' : 'block'};
  }

  &:hover {
    ${SectionHeader} {
      opacity: 1;
    }
  }
`
export const Section = styled(SectionFactory)<SectionProps>`


  & + & {
    border-top: solid 1px rgba(0,0,0, 0.1);
  }
`
