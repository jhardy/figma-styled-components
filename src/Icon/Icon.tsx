import * as React from 'react'
import styled from 'styled-components'
import * as IconGlyph from './components'
import { IconNames } from './IconNames'

export interface IconProps {
  name: IconNames
  color?: string
  size: number
}
const IconFactory: React.FC<IconProps> = ({ name, color, size, ...props }) => {
  const IconSVG = IconGlyph[name]
  return(
    <div {...props}>
      <IconSVG />
    </div>
  )
}

export const Icon = styled(IconFactory)`
  color: ${(props) => props.color || 'currentColor'};
  line-height: 0;
`
