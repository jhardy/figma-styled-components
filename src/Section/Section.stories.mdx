import { Section } from './Section'
import { Icon } from '../Icon'
import { Story, Meta } from '@storybook/addon-docs/blocks'

<Meta title='MDX|Section' component={Section} />

# Section

<Story name='Section'>
  <div className='component-box'>
    <Section title='Section 1'>Section Content</Section>
    <Section title='Collapsed Section' collapsed={true} actions={<Icon name='Plus' />}>
      Collapsed
    </Section>
  </div>
</Story>

<div className="props-table">

## Props

| Prop Name | Type    | Default | Required | Description                           |
| :-------- | :------ | :-----: | :------: | :------------------------------------ |
| title     | string  |    -    |   yes    | Title of the Section                  |
| actions   | JSX     |    -    |    -     | Icons to the right of a section title |
| collapsed | boolean |  false  |    -     | If the Section is collapsed or not    |

</div>
