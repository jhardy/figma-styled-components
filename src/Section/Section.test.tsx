import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Icon } from '../Icon'
import { Section } from './Section'



test('A default Section', () => {
  const tree = renderer.create(<Section title='Section'>Hello</Section>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Section with actions', () => {
  const tree = renderer.create(<Section title='Section' actions={<Icon name='Plus' />}>Hello</Section>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('A collapsed Section', () => {
  const tree = renderer.create(<Section title='Section' actions={<Icon name='Plus' />} collapsed={true}>Hello</Section>).toJSON()
  expect(tree).toMatchSnapshot()
})

