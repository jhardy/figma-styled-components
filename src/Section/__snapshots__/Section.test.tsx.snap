// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Section with actions 1`] = `
.c6 {
  color: currentColor;
  line-height: 0;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  color: rgba(0,0,0,0.8);
  background-color: #ffffff;
  font-weight: 600;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  padding: 8px;
}

.c2 .c3 {
  opacity: 1;
}

.c2 .c7 {
  display: block;
}

.c2:hover .c3 {
  opacity: 1;
}

.c1 + .c0 {
  border-top: solid 1px rgba(0,0,0,0.1);
}

<section
  className="c0 c1 c2"
  title="Section"
>
  <div
    className="c3 c4"
  >
    <span
      className="c5"
    >
      Section
    </span>
    <div
      className=""
    >
      <div
        className="c6"
      >
        <svg
          fill="none"
          height={32}
          width={32}
        >
          <path
            clipRule="evenodd"
            d="M15.5 15.5v-5h1v5h5v1h-5v5h-1v-5h-5v-1h5z"
            fill="currentColor"
            fillOpacity={0.8}
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="c7 "
  >
    Hello
  </div>
</section>
`;

exports[`A collapsed Section 1`] = `
.c6 {
  color: currentColor;
  line-height: 0;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  color: rgba(0,0,0,0.8);
  background-color: #ffffff;
  font-weight: 600;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 .c3 {
  opacity: 1;
}

.c8 .c7 {
  display: block;
}

.c8:hover .c3 {
  opacity: 1;
}

.c2 {
  padding: 8px;
}

.c2 .c3 {
  opacity: 0.5;
}

.c2 .c7 {
  display: none;
}

.c2:hover .c3 {
  opacity: 1;
}

.c1 + .c0 {
  border-top: solid 1px rgba(0,0,0,0.1);
}

<section
  className="c0 c1 c2"
  title="Section"
>
  <div
    className="c3 c4"
  >
    <span
      className="c5"
    >
      Section
    </span>
    <div
      className=""
    >
      <div
        className="c6"
      >
        <svg
          fill="none"
          height={32}
          width={32}
        >
          <path
            clipRule="evenodd"
            d="M15.5 15.5v-5h1v5h5v1h-5v5h-1v-5h-5v-1h5z"
            fill="currentColor"
            fillOpacity={0.8}
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="c7 "
  >
    Hello
  </div>
</section>
`;

exports[`A default Section 1`] = `
.c5 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  color: rgba(0,0,0,0.8);
  background-color: #ffffff;
  font-weight: 600;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  padding: 8px;
}

.c2 .c3 {
  opacity: 1;
}

.c2 .c6 {
  display: block;
}

.c2:hover .c3 {
  opacity: 1;
}

.c1 + .c0 {
  border-top: solid 1px rgba(0,0,0,0.1);
}

<section
  className="c0 c1 c2"
  title="Section"
>
  <div
    className="c3 c4"
  >
    <span
      className="c5"
    >
      Section
    </span>
  </div>
  <div
    className="c6 "
  >
    Hello
  </div>
</section>
`;
